main/src/ca/uwaterloo/flix/language/phase/unification/CofiniteIntSet.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2024 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.unification
+
+import scala.collection.immutable.SortedSet
+
+/**
+  * Represents a finite or co-finite set with an infinite universe of integers.
+  *
+  * All sets are either a [[SortedSet]] or a complement of it.
+  *
+  * No finite set is ever equivalent to universe.
+  */
+sealed trait CofiniteIntSet {
+
+  import CofiniteIntSet.{Compl, Set}
+
+  /** Returns `true` if `this` is [[CofiniteIntSet.universe]]. */
+  def isUniverse: Boolean = this match {
+    case Compl(s) if s.isEmpty => true
+    case Set(_) => false
+    case Compl(_) => false
+  }
+
+  /** Returns `true` if `this` is [[CofiniteIntSet.empty]]. */
+  def isEmpty: Boolean = this match {
+    case Set(s) if s.isEmpty => true
+    case Set(_) => false
+    case Compl(_) => false
+  }
+
```

Comment:
I think repeating " No finite set is ever equivalent to universe." would be good for both isUniverse and isEmpty.